crates/security/src/lib.rs

```diff
@@ -139,143 +83,41 @@ pub mod pallet {
     #[pallet::getter(fn active_block_number)]
     pub type ActiveBlockCount<T: Config> = StorageValue<_, T::BlockNumber, ValueQuery>;
 
+    #[pallet::storage]
+    pub type IsDeactivated<T: Config> = StorageValue<_, bool, ValueQuery>;
+
     #[pallet::pallet]
     pub struct Pallet<T>(_);
 
     // The pallet's dispatchable functions.
     #[pallet::call]
     impl<T: Config> Pallet<T> {
-        /// Set the parachain status code.
-        ///
-        /// # Arguments
-        ///
-        /// * `origin` - the dispatch origin of this call (must be _Root_)
-        /// * `status_code` - the status code to set
-        ///
-        /// # Weight: `O(1)`
+        /// Activate or deactivate active block counting.
         #[pallet::call_index(0)]
-        #[pallet::weight(<T as Config>::WeightInfo::set_parachain_status())]
-        #[transactional]
-        pub fn set_parachain_status(origin: OriginFor<T>, status_code: StatusCode) -> DispatchResultWithPostInfo {
+        #[pallet::weight(T::WeightInfo::activate_counter())]
+        pub fn activate_counter(origin: OriginFor<T>, is_active: bool) -> DispatchResult {
             ensure_root(origin)?;
-            Self::set_status(status_code);
-            Ok(().into())
-        }
 
-        /// Insert a new parachain error.
-        ///
-        /// # Arguments
-        ///
-        /// * `origin` - the dispatch origin of this call (must be _Root_)
-        /// * `error_code` - the error code to insert
-        ///
-        /// # Weight: `O(1)`
-        #[pallet::call_index(1)]
-        #[pallet::weight(<T as Config>::WeightInfo::insert_parachain_error())]
-        #[transactional]
-        pub fn insert_parachain_error(origin: OriginFor<T>, error_code: ErrorCode) -> DispatchResultWithPostInfo {
-            ensure_root(origin)?;
-            Self::insert_error(error_code)?;
-            Ok(().into())
-        }
+            IsDeactivated::<T>::set(is_active);
```

Comment:
This is slightly confusing, I would assume `is_active=true` enables counting but it does the opposite. I think we should rename the storage item, the extrinsic or the parameter.